rest_framework_jwt/views.py

```diff
@@ -106,6 +109,23 @@ class RefreshJSONWebToken(JSONWebTokenAPIView):
     serializer_class = RefreshJSONWebTokenSerializer
 
 
+class DeviceLogout(DestroyAPIView):
+    """
+    Logout user by deleting Device.
+    """
+    queryset = Device.objects.all()
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return self.queryset.filter(user=self.request.user)
+
+    def get_object(self):
```

Comment:
Why two? You have get_queryset and get object. I'd use only get object, there is no point in keeping two functions.